mqtt-pushgateway.py

```diff
@@ -124,13 +125,23 @@ def http_metrics():
     return Response('\n'.join(content + ['']), mimetype="text/plain")
 
 
-def on_message(client, userdata, message):
+def update_topic(topic, message):
     try:
-        metrics[message.topic].update(message.topic, message.payload)
+        metrics[topic].update(topic, message)
     except:
         logging.warning("Value is neither numeric nor valid utf-8, ignored", exc_info=True)
     else:
-        logging.info("Message received: %s => %s", message.topic, message.payload)
+        logging.info("Message received: %s => %s", topic, message)
+
+
+def on_message(client, userdata, message):
+    try:
+        json_message = json.loads(message.payload)
+        for k, v in json_message.items():
+            topic = "{}/{}".format(message.topic, k)
+            update_topic(topic, v)
+    except (ValueError):
+        update_topic(message.topic, message.payload)
```

Comment:
I'd actually prefer to keep this in one function, given the way I just refactored the `on_message` function. I think the "try json" code should now live between the [data decoding](https://github.com/Svedrin/mqtt-pushgateway/blob/53308696da4c6a268a56f5814af97ab0b4834212/mqtt-pushgateway.py#L128-L134) and [topic update](https://github.com/Svedrin/mqtt-pushgateway/blob/53308696da4c6a268a56f5814af97ab0b4834212/mqtt-pushgateway.py#L136-L139) parts.